nilearn/surface/surface.py

```diff
@@ -1975,3 +1976,243 @@ def new_img_like(ref_img, data):
+    # We use this as a weighting to make sure that
+    # smoothing takes into account the distance between each vertex neighbor
+    if values in ("len", "invlen"):
+        coords = mesh.coordinates
+        edge_lens = np.sqrt(np.sum((coords[u, :] - coords[v, :]) ** 2, axis=1))
+        if dtype is None:
+            dtype = edge_lens.dtype
+        else:
+            edge_lens = edge_lens.astype(dtype)
+        if values == "invlen":
+            edge_lens = 1 / edge_lens
+    elif values == "ones":
+        if dtype is None:
+            edge_lens = np.ones_like(edges)
+        else:
+            edge_lens = np.ones(edges.shape, dtype=dtype)
+
+    # We can now make a sparse matrix.
+    ee = np.concatenate([edge_lens, edge_lens])
+    uv = np.concatenate([u, v])
+    vu = np.concatenate([v, u])
+
+    return csr_matrix((ee, (uv, vu)), shape=(n, n))
+
+
+def smooth_img(
+    imgs,
+    iterations=1,
+    distance_weights=False,
+    vertex_weights=None,
+    center_surround_knob=0,
+    match="sum",
+):
+    """Smooth values along the surface.
+
+    Parameters
+    ----------
+    imgs : SurfaceImage
+        The surface whose is to be smoothed.
+        In the case of 2D data, each sample is smoothed independently.
+
+    iterations : :obj:`int`, default = 1
+        The number of times to repeat the smoothing operation
+        (it must be a positive value).
+        Defaults to 1
+
+    distance_weights : :obj:`bool`, default = False
+        Whether to add distance-based weighting to the smoothing.
+        With such weights, the value calculated for each vertex
+        at each iteration is the weighted sum of neighboring vertices
+        where the weight on each neighbor is the inverse
+        of the distances to it.
+
+    vertex_weights : SurfaceImage or None, default = None
+        A SurfaceImage whose data are vector of weights, one per vertex.
+        These weights are normalized and
+        applied to the smoothing
+        after the application of center-surround weights.
+
+    center_surround_knob : :obj:`float`, default = 0
+        The relative weighting of the center and the surround
+        in each iteration of the smoothing.
+        If the value of the knob is `k`,
+        then the total weight of vertices that are neighbors
+        of a given vertex (the vertex's surround)
+        is `2**k` times the weight of the vertex itself (the center).
+        A value of 0 (the default) means that, in each smoothing iteration,
+        each vertex is updated with the average of its value
+        and the average value of its neighbors.
+        A value of `-inf` results in no smoothing because the entire
+        weight is on the center, so each vertex is updated with its own value.
+        A value of `inf` results in each vertex being updated
+        with the average of its neighbors without including its own value.
+
+    match : { 'sum', 'mean', 'var', 'dist', None}, default = "sum"
```

Comment:
similarly: is this type of normalization needed?
We do no such thing for volume smoothing: why do it here? Am I missing something?